kotlinx-coroutines-core/js/test/PropagateExceptionFinalResortTest.kt

```diff
@@ -0,0 +1,45 @@
+package kotlinx.coroutines
+
+import kotlinx.coroutines.testing.*
+import kotlin.js.*
+import kotlin.test.*
+
+class PropagateExceptionFinalResortTest : TestBase() {
+    @BeforeTest
+    fun setupExceptionListenerOnly() {
+        // Remove a Node.js's internal listener, which prints the exception to stdout.
+        js("""
+            globalThis.originalListeners = process.listeners('uncaughtException');
+            process.removeAllListeners('uncaughtException');
+        """)
+    }
+
+    @AfterTest
+    fun restoreListeners() {
+        js("""
+            if (globalThis.originalListeners) {
+                process.removeAllListeners('uncaughtException');
+                globalThis.originalListeners.forEach(function(listener) {
+                    process.on('uncaughtException', listener);
+                });
+            }
+        """)
+    }
+
+    @Test
+    fun testThrows() = runTest {
```

Comment:
I'd expect a more descriptive name, like `testHandlingUncaughtExceptionsInNodeJs`. Maybe even a documentation comment explaining what exactly we are checking here. Given how much JS code there is in this file, it's not trivial to figure out.

Reply:
It's just that all the info is in the class name + path (for platform), and naming this accurately would be duplicating those. Had we had a dozen tests, we wouldn't be naming them so verbosely. I've done it anyways for now.